dds/DCPS/DomainParticipantFactoryImpl.cpp

```diff
@@ -100,7 +100,7 @@ DomainParticipantFactoryImpl::create_participant(
   // needed for the built in topics during enable.
   dp->set_object_reference(dp_obj);  //xxx no change
 
-  if (qos_.entity_factory.autoenable_created_entities == 1) {
+  if (qos_.entity_factory.autoenable_created_entities == true) {
```

Comment:
That will be at least 300 places where another set of code changes are necessary, let us do that in another PR